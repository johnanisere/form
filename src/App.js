import React from "react";
import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";

import storage from "./store";
import Loading from "./components/Loader";
import Opportunity from "./compositions/Opportunity";
import ToastContainer from "./components/ToastContainer";
import ErrorBoundary from "./compositions/ErrorBoundary";

const { store, persistor } = storage();

const App = () => (
  <ErrorBoundary>
    <ToastContainer />
    <Provider store={store}>
      <PersistGate loading={<Loading />} persistor={persistor}>
        <Opportunity />
      </PersistGate>
    </Provider>
  </ErrorBoundary>
);

export default App;
