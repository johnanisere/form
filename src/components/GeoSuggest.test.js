import React from "react";
import { shallow } from "enzyme";

import App from "./GeoSuggest";

it("App component exists", () => {
  const parent = shallow(<App />);
  expect(parent.exists()).toBe(true);
});

it("App component renders correctly", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
