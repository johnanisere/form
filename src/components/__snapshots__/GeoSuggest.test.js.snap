// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoActivateFirstSuggest": false,
      "disabled": false,
      "fixtures": Array [],
      "getSuggestLabel": [Function],
      "highlightMatch": true,
      "ignoreEnter": false,
      "ignoreTab": false,
      "initialValue": "",
      "maxFixtures": 10,
      "minLength": 1,
      "onKeyDown": [Function],
      "onKeyPress": [Function],
      "onSuggestSelect": [Function],
      "placeholder": "City",
      "queryDelay": 250,
      "skipSuggest": [Function],
      "style": Object {},
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoActivateFirstSuggest": false,
        "disabled": false,
        "fixtures": Array [],
        "getSuggestLabel": [Function],
        "highlightMatch": true,
        "ignoreEnter": false,
        "ignoreTab": false,
        "initialValue": "",
        "maxFixtures": 10,
        "minLength": 1,
        "onKeyDown": [Function],
        "onKeyPress": [Function],
        "onSuggestSelect": [Function],
        "placeholder": "City",
        "queryDelay": 250,
        "skipSuggest": [Function],
        "style": Object {},
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
